src/transaction_builder.js

```diff
@@ -210,8 +212,18 @@ function prepareInput (input, kpPubKey, redeemScript, hashType) {
     // pay-to-scriptHash is not possible without a redeemScript
     if (input.prevOutType === 'scripthash') throw new Error('PrevOutScript is P2SH, missing redeemScript')
 
+    if (!input.pubKeys || input.pubKeys.length === 0) {
+      // maybe we can derive more information about the script
+      // now that we have a kpPubKey?
+      expanded = expandOutput(input.prevOutScript, input.prevOutType, kpPubKey)
+      input.pubKeys = expanded.pubKeys
+      input.signatures = expanded.signatures
```

Comment:
@runn1ng specifically,  this step was missing and is a simple result of you gave `TransactionBuilder` more information than usual (at the start),  but not enough to know if it could sign with `kpPubKey`... until you went to `sign` with it,  but just assumed that if `prevOutScript` was defined,  that it was "complete".
Which isn't true... and also opens up an interesting can of worms around what else could be "added" or "derived"... but this will do for now.